src/utils/errors/ws.ts

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable max-classes-per-file */
+
+/**
+ * Thrown when a WebSocket request does not receive a response within the configured timeout period.
+ *
+ * @property {string} name - The name of the error, always 'TimeoutError'.
+ */
+export class TimeoutError extends Error {
```

Comment:
Should `TimeoutError` and `WebSocketNotConnectedError` inherit from `CustomError/LibraryError/starknetError`?

Reply:
https://github.com/starknet-io/starknet.js/pull/1416/commits/aea7f527eb4a881bc5172e511c96abd2affd36d5